db/migrate/20161127171855_create_projects.rb

```diff
@@ -0,0 +1,12 @@
+class CreateProjects < ActiveRecord::Migration[5.0]
+  def change
+    create_table :projects do |t|
+      t.string :repo_url
+      t.string :live_preview
+      t.references :user, foreign_key: true
```

Comment:
We could avoid this over complication by just calling the columns for user and lesson `user_id` and `lesson_id` rails will then make those foreign keys automatically.